sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Canonicalize.scala

```diff
@@ -80,6 +80,13 @@ object Canonicalize {
       orderCommutative(a, { case And(l, r) if l.deterministic && r.deterministic => Seq(l, r)})
         .reduce(And)
 
+    case o: BitwiseOr =>
+      orderCommutative(o, { case BitwiseOr(l, r) => Seq(l, r) }).reduce(BitwiseOr)
```

Comment:
should we check whether both branches are deterministic like or/and in the above?

Reply:
Bitwise operations will always execute both sides. Boolean AND and OR can short circuit and not execute one side.
The bitwise operations are more like + and * in this regard.

Reply:
OK. I was thinking more about side-effect from non-deterministic expressions (e.g., println), and the evaluation order matters in this case, but it seems other existing transformations don't handle this as well.

Reply:
Canonicalized expression is only used for determining if two expressions have same result. We don't run it directly. And we usually consider `deterministic` together when using `canonicalized`, e.g. `semanticEquals`.